### PR TITLE
Add version range support in web UI

### PR DESCRIPTION
**Pull request by bclozel**
This commit introduces zepto.js as a minimal front-end library to
manipulate the DOM and a lightweight library `Versions` to parse and
compare version ranges (OSGI style).

With this change, only Boot starter dependencies that are compatible
with the selected Boot version should be displayed in the UI.

Partial fix for #62

### CODE REVIEW

File: initializr/src/main/resources/templates/home.html [hunk 1]
```diff
@@ -111,13 +111,13 @@
...
             </div>
-            <div class="col-sm-6">
+            <div id="dependencies" class="col-sm-6">
                 <h3>Project dependencies</h3>
                 <% dependencies.each { %>
                 <div class="form-group col-xs-6">
                 <h4>${it.name}</h4>
                     <% it.content.each { %>
-                    <div class="checkbox">
+                    <div class="checkbox" data-range="1.2.1.RELEASE">
```
Comment (bclozel) [inline initializr/src/main/resources/templates/home.html]: This should be fixed with the real value coming from boot metadata. Leaving the value empty should be fine as well.

**Comment by snicoll** [comment]
@wilkinsona managed to break it :) Select Jersey, then switch to Boot 1.1.x (jersey will disappear as it should). Click generate project and it will fail because jersey is still checked. I guess we need to add an additional script that uncheck anything that's not valid against the new boot version.

**Comment by snicoll** [comment]
So the tests are breaking even if I simulate with Chrome which is a bit surprising. It seems the usage of `getComputedStyle` is a bad idea?

**Comment by bclozel** [comment]
Just updated this PR to uncheck input checkboxes when hiding dependencies.

**Comment by bclozel** [comment]
This PR currently fails integration tests.
Before [version 2.14](http://htmlunit.sourceforge.net/changes-report.html#a2.14), HtmlUnit doesn't support [window.getComputedStyle](https://developer.mozilla.org/en-US/docs/Web/API/Window.getComputedStyle), so tests fail with:

```

createDefaultGroovyProject(io.spring.initializr.web.MainControllerFormIntegrationTests)  Time elapsed: 0.085 sec  <<< ERROR!
com.gargoylesoftware.htmlunit.ScriptException: ReferenceError: "getComputedStyle" is not defined (http://localhost/js/zepto.min.js#2)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine$HtmlUnitContextAction.run(JavaScriptEngine.java:684)
```

When upgrading this app to htmlunit 2.14 (and using a compatible webclient, so basically Chrome, Firefox, IE9+), tests still fail with:

````
Caused by: java.lang.NullPointerException: null
	at com.gargoylesoftware.htmlunit.javascript.host.Window.getComputedStyle(Window.java:1613)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
````

It looks like htmlunit explicitely supports some libraries such as jQuery and others, but does not mention zeptojs. I've tried using the non-minified version as well but ended with the same results.

We could use jQuery to avoid this problem, but it's x3 bigger and brings no additional value here.

**Comment by snicoll** [comment]
merged, thanks!

**Pull request closed without merging**